scapy/utils.py

```diff
@@ -429,17 +429,22 @@ class ContextManagerSubprocess(object):
     >>>     subprocess.Popen(["unknown_command"])
 
     """
-    def __init__(self, name):
+    def __init__(self, name, prog, _raise=True):
         self.name = name
+        self.prog = prog
+        self._raise = _raise
 
     def __enter__(self):
         pass
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if exc_type == OSError:
-            msg = "%s: executing %r failed"
-            log_runtime.error(msg, self.name, conf.prog.wireshark, exc_info=1)
-            return True  # Suppress the exception
+        if isinstance(exc_value, (OSError, TypeError)):
+            msg = "%s: executing %r failed" % (self.name, self.prog) if self.prog else "Could not execute %s, is it installed ?" % self.name
+            if self._raise:
```

Comment:
Would it make sense here to use `conf.interactive` (see #791)?

Reply:
Good idea. Updated & Rebased